src/com/amplitude/api/DeviceInfo.java

```diff
@@ -145,11 +148,17 @@ private String getCountryFromLocation() {
         private String getCountryFromNetwork() {
             TelephonyManager manager = (TelephonyManager) context
                     .getSystemService(Context.TELEPHONY_SERVICE);
-            if (manager.getPhoneType() != TelephonyManager.PHONE_TYPE_CDMA) {
-                String country = manager.getNetworkCountryIso();
-                if (country != null) {
-                    return country.toUpperCase(Locale.US);
+            try {
+                if (manager.getPhoneType() != TelephonyManager.PHONE_TYPE_CDMA) {
+                    String country = manager.getNetworkCountryIso();
+                    if (country != null) {
+                        return country.toUpperCase(Locale.US);
+                    }
                 }
+            } catch (NullPointerException e) {
```

Comment:
I hesitate to say this, but I think catching all Exceptions would make sense here, as it's a system call and not much we can do about it. It should probably also include the getSystemService call on line 149.